### PR TITLE
 Convert linfa-preprocessing Benchmark to Criterion #274 

### PR DESCRIPTION
**Pull request by lucas-montes**
I currently only started with whitening to check if the changes are according to what is expected

### CODE REVIEW

File: algorithms/linfa-preprocessing/benches/whitening_bench.rs [hunk 1]
```diff
@@ -1,38 +1,47 @@
...
+        (Whitener::pca(), "pca"),
+    ] {
+        for nfeatures in (10..100).step_by(10) {
+            let dataset = make_dataset(size, nfeatures, 1, feat_distr, target_distr);
+            benchmark.bench_function(
+                BenchmarkId::new(fn_name, format!("{}x{}", nfeatures, size)),
+                |bencher| {
+                    bencher.iter(|| {
+                        whitener
+                            .fit(black_box(&dataset.clone()))
```
Comment (YuhanLiin) [inline algorithms/linfa-preprocessing/benches/whitening_bench.rs:28]: unnecessary clone

File: algorithms/linfa-preprocessing/benches/whitening_bench.rs [hunk 2]
```diff
@@ -1,38 +1,47 @@
...
+                            .unwrap()
+                            .transform(black_box(dataset.clone()));
```
Comment (YuhanLiin) [inline algorithms/linfa-preprocessing/benches/whitening_bench.rs:30]: I think you can put `dataset.view()` here to prevent unnecessary clone

Comment (lucas-montes) [reply algorithms/linfa-preprocessing/benches/whitening_bench.rs]: Done

File: algorithms/linfa-preprocessing/benches/whitening_bench.rs (hunk 1 above)
Comment (lucas-montes) [reply algorithms/linfa-preprocessing/benches/whitening_bench.rs]: Done

**Review by YuhanLiin — APPROVED** [review]

**Pull request merged**